### PR TITLE
Fix build image, add target for running `make test` and `make build` inside build image

### PR DESCRIPTION
**Pull request by errordeveloper**
xref #1461

### CODE REVIEW

File: Makefile.docker [hunk 1]
```diff
@@ -73,3 +78,11 @@ eksctl-image: intermediate-image ## Build the eksctl image that has release arte
 	      --iidfile=.eksctl_image.iid \
 	      --tag="$(eksctl_image_name)" -
 
+.PHONY: test build
+build test: ## Build and test using the build image
```
Comment (martina-if) [inline Makefile.docker]: test build or build test?

File: Makefile.docker [hunk 2]
```diff
@@ -22,19 +23,23 @@ update_build_image_manifest = git ls-tree --full-tree @ -- $(build_image_input)
...
+	  || git commit --quiet .build_image_manifest --message 'Update build image manifest'
 	@# There is no portable way of using sed -i on BSD and GNU, so we use temporary files instead
 	for i in .github/workflows/commit.yaml .circleci/config.yml ; do \
-	  sed "s|\(weaveworks/eksctl-build\):.*$$|\1:$(build_image_tag)|" "$${i}" > "$${i}.sedtmp" && mv "$${i}.sedtmp" "$${i}" ; \
+	  sed "s|\(weaveworks/eksctl-build\):.*$$|\1:$$($(build_image_tag_cmd))|" "$${i}" > "$${i}.sedtmp" && mv "$${i}.sedtmp" "$${i}" ; \
 	done
-	git commit --quiet .build_image_manifest .github/workflows/commit.yaml .circleci/config.yml --message 'Update build image'
-	$(info build image name is $(build_image_name))
+	git diff --quiet -- .github/workflows/commit.yaml .circleci/config.yml \
+	  || git commit --quiet .github/workflows/commit.yaml .circleci/config.yml --message 'Update build image in CI'
```
Comment (martina-if) [inline Makefile.docker]: Perhaps say something like "Please update build image in CI with make...."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -84,6 +84,20 @@ make integration-test-container TEST_V=1
...
+
+```
+make -f Makefile.docker test
+make -f Makefile.docker build
+```
+
+> NOTE: It is not the most convenient way of working on the project, as
+> binaries are built inside the container and cannot be tested manually,
+> also majority of end-users consume binaries and not Docker images.
+> It is recommended to use `make build` etc, unless there is issue in CI
```
Comment (martina-if) [inline CONTRIBUTING.md]:
```suggestion
> It is recommended to use `make build` etc, unless there is an issue in CI
```

File: Makefile.docker (hunk 1 above)
Comment (martina-if) [reply Makefile.docker]: If this is correct go ahead and merge

Comment (errordeveloper) [reply Makefile.docker]: The order doesn't matter here, but I'll make it same the line above for cosmetic reasons.

**Pull request merged**